File: Java/Easy/Accepted/shuffle-string_5.java
class Solution {
    public String restoreString(String s, int[] indices) {
        
        char[] chars = s.toCharArray();
        char[] result = new char[indices.length];
        for (int i = 0; i < indices.length; i++) {
            result[indices[i]] = chars[i];
        }
        return new String(result);
    }
}

File: Rust/Easy/RuntimeError/shuffle-string_5.rs
impl Solution {
    pub fn restore_string(s: String, indices: Vec<i32>) -> String {
        
        let mut v: Vec<char> = s.chars().collect();
        let mut s = String::new();
        for i in 0..indices.len() {
            v[indices[i] as usize] = s.chars().nth(i).unwrap();
        }
        for i in 0..v.len() {
            s.push(v[i]);
        }
        s
    }
}